orchagent/neighorch.cpp

```diff
@@ -39,6 +50,15 @@ bool NeighOrch::addNextHop(const IpAddress &ipAddress, const string &alias)
     }
 
     NextHopKey nexthop = { ipAddress, alias };
+    if(m_intfsOrch->isRemoteSystemPortIntf(alias))
+    {
+        //For remote system ports kernel nexthops are always on inband. Change the key
+        Port inbp;
+        if(gPortsOrch->getInbandPort(inbp))
```

Comment:
Is it possible that getInbandPort may fail? Do we need to handle this case?

Reply:
This neighbor is on the remote system ports. Remote system port neighbors will never come here without Inband interface already up and active. So this getInbandPort() for remote system port neighbors will never fail.

Reply:
If getInbandPort() will never fail, can we use assert here?

Reply:
Yes. We'll use assert here.